src/tools/wpa/QuicEventDataSource.cs

```diff
@@ -17,15 +18,22 @@ namespace QuicEventDataSource
         "QUIC Event",
         "Events generated by MsQuic")]
     [FileDataSource("etl", "Event Trace Log")]
+#pragma warning disable CA1724 // Class name should not match namespace name
     public class QuicEventDataSource : CustomDataSourceBase
+#pragma warning restore CA1724 // Class name should not match namespace name
     {
-        IApplicationEnvironment applicationEnvironment;
+        private IApplicationEnvironment? applicationEnvironment;
 
         protected override ICustomDataProcessor CreateProcessorCore(
             IEnumerable<IDataSource> dataSources,
             IProcessorEnvironment processorEnvironment,
             ProcessorOptions options)
         {
+            if (applicationEnvironment == null)
+            {
+                throw new InvalidOperationException("SetApplicationEnvironmentCore must be called before CreateProcessorCore");
+            }
```

Comment:
Why are you adding this?

Reply:
Because this variable is nullable, since its set after construction. So at this point, it can be null. But the constructor we're passing it into requires the variable to be non null. It would be broken if that happened as well, as late calling the set function would do nothing.

Reply:
Well, don't make it nullable. I don't want to add extra error checking code to this project. I want it streamlined.